### PR TITLE
Move group sparse covariance

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Trying to Fix Issue #850 

See #850 for more details.
1. Moved file ```group_sparse_covariance``` to ```nilearn.connectome```
2. Moved test files
3. Changed doc and examples accordingly.

### CODE REVIEW

File: doc/connectivity/connectome_extraction.rst [hunk 1]
```diff
@@ -138,7 +138,7 @@ And it provides one estimated covariance and inverse-covariance
 
 |
 
-.. currentmodule:: nilearn.group_sparse_covariance
+.. currentmodule:: nilearn.connectome.group_sparse_covariance
```
Comment (GaelVaroquaux) [inline doc/connectivity/connectome_extraction.rst]: I think that you should be able to have simply nilearn.connectome here, given that GroupSparseCovariance is imported in nilearn.connectome.__init__. Isn't it the case?

File: doc/developers/group_sparse_covariance.rst [hunk 2]
```diff
@@ -5,7 +5,7 @@ Group-sparse covariance estimation
 ..
     Explain which case is implemented (p=2, unpenalized diagonal)
 
-.. currentmodule:: nilearn.group_sparse_covariance
+.. currentmodule:: nilearn.connectome.group_sparse_covariance
```
Comment (GaelVaroquaux) [inline doc/developers/group_sparse_covariance.rst]: Same remark about nilearn.connectome without the group_sparse_covariance.

**Comment by GaelVaroquaux** [comment]
Overall this looks great, aside the small comments that I made.

**Comment by KamalakerDadi** [comment]
Fixed @GaelVaroquaux comments

**Comment by GaelVaroquaux** [comment]
LGTM. Merging. Thanks!

**Pull request merged**